urlshort/handler.go

```diff
@@ -33,5 +42,24 @@ func MapHandler(pathsToUrls map[string]string, fallback http.Handler) http.Handl
 // a mapping of paths to urls.
 func YAMLHandler(yml []byte, fallback http.Handler) (http.HandlerFunc, error) {
 	// TODO: Implement this...
-	return nil, nil
+	parsedYaml, err := parseYAML(yml)
+	if err != nil {
+		return nil, err
+	}
+	pathMap := buildMap(parsedYaml)
+	return MapHandler(pathMap, fallback), nil
+}
+
+func parseYAML(yml []byte) (out []map[string]string, err error) {
+	err = yaml.Unmarshal(yml, &out)
+	return out, err
+}
+
+func buildMap(parsedYaml []map[string]string) map[string]string {
+	pathMap := make(map[string]string)
+	for _, entry := range parsedYaml {
+		key := entry["path"]
+		pathMap[key] = entry["url"]
```

Comment:
It looks like you could have used a struct instead of a map here. Using a struct would be good practice for using go tags.